### PR TITLE
Reimplement executeCommand

### PR DESCRIPTION
**Pull request by jwmwalrus**
## Contribution type

*Bug fix*

## Description of change

Although pull request #229 solves the issue of builds never finishing  when using Codeception (issue #217), it does so in a sub-optimal way.

In general, the issue is that when Codeception finishes, *stdout* remains open, so PHP Censor assumes the process is still running.

Taking into account that PHP Censor already lists Symfony's Process component as a dependency, I reimplemented the `executeCommand` method to use said component for collecting the output of the process being executed, while checking periodically if the process is still running.

Symfony's Process default timeout is 60 seconds, so I changed it to a 86400, just to be sure it won't be an issue.  There's also an *idle* timeout, which I set to 600 seconds.

Codeception doesn't seem to use *stderr*, so in order to detect actual failure in the testing phase, the number of failed tests is used.

### CODE REVIEW

**Comment by corpsee** [comment]
@jwmwalrus Thank you for the great work! I will review request as soon as possible. :+1:

**Comment by corpsee** [comment]
@ptejada Thank you for the review!

File: src/Helper/CommandExecutor.php [hunk 1]
```diff
@@ -79,29 +89,45 @@ public function executeCommand($args = [])
...
+                break;
+            }
+        }
 
-        $lastOutput = '';
-        $lastError  = '';
-        if (is_resource($process)) {
-            fclose($pipes[0]);
+        $process = new Process($command, $this->buildPath);
+        $process->setTimeout(86400);
```
[the remark this answers is not in the document]
Comment (jwmwalrus) [reply src/Helper/CommandExecutor.php]: As I mentioned in the PR comment, Process' default is 60 seconds ---i.e., an exception will be thrown if the execution of any command takes more than 60 seconds.

**Comment by corpsee** [comment]
@jwmwalrus :+1:

**Comment by corpsee** [comment]
The PR merged to `master` branch and would be released with next minor version `1.1.0`.

**Comment by corpsee** [comment]
@jwmwalrus The PR has bug with env variables. Symfony's Process don't inherit parent process env variables. See: #261.

I will revert the PR before we haven't fix.

**Pull request merged**